### PR TITLE
Collision detection

### PR DESCRIPTION
**Pull request by si74**

### CODE REVIEW

File: src/Player.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    // connects our texture with dest to control position
+    SDL_QueryTexture(playerTex, NULL, NULL, &dest.w, &dest.h);
+
+    // adjust height and width of our image box.
+    dest.w /= 1.5;
+    dest.h /= 1.5;
+
+    // sets initial x-position of object middle of screen
+    dest.x = _x;//(SCREEN_WIDTH - dest.w) / 2;
```
Comment (si74) [inline src/Player.cpp]: can we remove some of the commented out code?

File: src/Game.cpp [hunk 2]
```diff
@@ -107,73 +88,40 @@ void Game::handleEvents() {
...
 
-    //SDL_Rect block1;
-    //renderBlock(window, &block1); // TODO render background function that renders all the blocks, also sets boundaries for player
-    map->DrawMap();
-    SDL_RenderCopy(renderer, playerTex, NULL, &dest); // player
-    //SDL_RenderCopy(renderer, map->water, NULL, &(map->dest));
+   //map->DrawMap();
+
+    //render game objects
+    player1->render(renderer); // player
```
Comment (si74) [inline src/Game.cpp:128]: nit: remove the comment for //player

File: src/Game.cpp [hunk 3]
```diff
@@ -107,73 +88,40 @@ void Game::handleEvents() {
...
 }
 
 
 void Game::update() {
     count++;
-    destRect.h = 128;
-    destRect.w = 128;
-
-    //map->LoadMap(); todo text file with different arrays
+    // destRect.h = 128;
```
Comment (si74) [inline src/Game.cpp:115]: can we remove this commented out code as well?

File: src/Game.cpp [hunk 4]
```diff
@@ -107,73 +88,40 @@ void Game::handleEvents() {
...
-
-    // left boundary 
-    if (dest.x < 0) {
-        dest.x = 0; 
-    }
+            if (event.key.keysym.scancode == SDL_SCANCODE_ESCAPE) { // exit game
+                isRunning = false;
+                break;
+            }
+            player1->move(event); // player
```
Comment (si74) [inline src/Game.cpp]: remove // comment

File: src/Ball.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        dest.x = SCREEN_WIDTH - dest.w; // TODO game over
+    } else if (dest.x < 0) { // left boundary
+        dest.x = 0; 
+    }
+
+    //hits player paddle
+    else if (SDL_HasIntersection(player_rect, &dest)) {
+        velocity_x *= -1;
+        velocity_y = rand() % ANGLE_RANGE - (ANGLE_RANGE-1);
+        // velocity_y =  dest.y - player_rect->y;
```
Comment (si74) [inline src/Ball.cpp]: remove this commented out code as well

**Review by yct4 — APPROVED** [review]

File: src/Ball.cpp [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    // sets initial y-position of object middle of screen
+    dest.y = (SCREEN_HEIGHT - dest.h) / 2;
+
+    // speed of ball
+    velocity_x = -2;
+    velocity_y = 0;
+}
+
+void Ball::move(const SDL_Rect* player_rect) {
+    // dest.y += speed / 50;
```
Comment (si74) [inline src/Ball.cpp]: should we remove this line?

File: src/Player.cpp [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+    // sets initial y-position of object middle of screen
+    dest.y = _y;//(SCREEN_HEIGHT - dest.h) / 2;
+
+    // keybindings
+    // scancode_up = _up;
```
Comment (si74) [inline src/Player.cpp]: remove this as well?

File: src/Player.cpp [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+void Player::move(SDL_Event& event) {
+    // keyboard API for key pressed
+    if (event.key.keysym.scancode == scancode_up) {
+        dest.y -= speed / 10;
+    } else if (event.key.keysym.scancode == scancode_down) {
+        dest.y += speed / 10;
+    }
+
+    // dest.y += speed / 30;
```
Comment (si74) [inline src/Player.cpp]: also remove this?

**Review by si74 — COMMENTED** [review]
👍

**Pull request merged**